src/libs/HttpUtils.js

```diff
@@ -28,10 +30,30 @@ function processHTTPRequest(url, method = 'get', body = null) {
  * @returns {Promise}
  */
 function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure = false) {
+    if (command !== 'Log') {
+        info('Making API request', false, {
```

Comment:
> Is this the right place for this? Seems like another circular dependency since info() is ultimately calling API.Log() which is calling xhr().

Yes? The circular dependency is kind of a lie, since the dependency is not used on load (as shown by this hack)... 

> If we want to do something before and after the requests happen then we could maybe hook this into Network.js similar to how we handle the error or success of all requests...
> But maybe add something like a onBeforeRequest and the ability to add more than one onResponse callback?

Oh that could be better, maybe I should add onBeforeRequest and onAfterRequest which would run in the promise's `finally` (so that we can log failed requests too, that I just realize we are not in the current code). Sounds good?

Reply:
Yep it would be really good to have information about when the error is thrown. I think we maybe only have this for unhandled errors in the error boundary. I think failed API calls just get retried maybe. It's been a while since I've looked into how we are handling it but probably not as best we could.

Reply:
Crap! If I add Log to API then I have a circular dependency :rage1: 
Any tips on how to avoid that?

Reply:
Dang, without doing a deeper dive - one quick solution I can think of is to handle it the same way that we've done in `Network.js`.

Something like, `HttpUtils.onBeforeRequest()` and on `HttpUtils.onAfterRequest()`

Then pass all the data about the request to the callback and have whoever "registers" the callback decide what to do i.e. call `Log.info()`, `API.Log`, or whatever.

Reply:
Oh wait we already talked about this above... why didn't it work? I don't get it.

Reply:
Maybe we can set up the callbacks someplace else besides `API`?

Reply:
hahaha yes we did. Check the changes I did that I think is what you were suggesting in this PR https://github.com/Expensify/App/pull/4374 (it's targeting this one, feel free to change the base branch if that makes it easier for you).
Basically API requires log and log API... but maybe I misunderstood what you were saying.

Reply:
Can we register the callbacks in another file besides `API` ?

Reply:
hmmm I guess we can? Where should I?

Reply:
Maybe in App?

Reply:
Although there's some things already tied to that file... like isAuthenticating, addDefaultValuesToParameters, credentials 🤔

Reply:
I can also leave it as it is here 😬

Reply:
Waiting on your confirmation

Reply:
Maybe here in `App.js` ? 

https://github.com/Expensify/App/blob/main/src/App.js

Or an actions file... I think basically anything that is not `API`, `Network`, or `HttpUtils` would work.

Reply:
If I move it there, I need to move half of the stuff there... handleExpiredAuthToken, credentials, addDefaultValuesToParameters, isAuthenticating I don't think I should move all those things to App.

Reply:
Ok, I guess we can worry about this another time? Feels like we took a wrong turn somewhere, but if there are that many changes then it's out of scope. Mind if I create a follow up issue to maybe think about it this some more?

Reply:
Not at all, go ahead. 

So, does the PR looks good? I need to retest it and update the other PRs too.